RadixWallet/Features/PreAuthorizationReview/PreAuthorizationReview+View.swift

```diff
@@ -0,0 +1,268 @@
+		let sliderResetDate: Date
+		let expiration: Expiration?
+		let secondsToExpiration: Int?
+		let globalControlState: ControlState
+		let sliderControlState: ControlState
+		let showRawManifestButton: Bool
+
+		var dAppName: String? {
+			dAppMetadata?.name?.rawValue
+		}
+	}
+
+	@MainActor
+	struct View: SwiftUI.View {
+		let store: StoreOf<PreAuthorizationReview>
+
+		@SwiftUI.State private var showNavigationTitle = false
+
+		private let coordSpace: String = "PreAuthorizationReviewCoordSpace"
+		private let navTitleID: String = "PreAuthorizationReview.title"
+		private let showTitleHysteresis: CGFloat = .small3
+
+		var body: some SwiftUI.View {
+			WithViewStore(store, observe: \.viewState, send: { .view($0) }) { viewStore in
+				content(viewStore)
+					.controlState(viewStore.globalControlState)
+					.background(.app.white)
+					.toolbar {
+						ToolbarItem(placement: .principal) {
+							if showNavigationTitle {
+								navigationTitle(dAppName: viewStore.dAppName)
+							}
+						}
+					}
+					.onAppear {
+						store.send(.view(.appeared))
+					}
+					.destinations(with: store)
+			}
+		}
+
+		private func navigationTitle(dAppName: String?) -> some SwiftUI.View {
+			VStack(spacing: .zero) {
+				Text(L10n.PreAuthorizationReview.title)
+					.textStyle(.body2Header)
+					.foregroundColor(.app.gray1)
+
+				if let dAppName {
+					Text(L10n.PreAuthorizationReview.subtitle(dAppName))
+						.textStyle(.body2Regular)
+						.foregroundColor(.app.gray2)
+				}
+			}
+		}
+
+		private func content(_ viewStore: ViewStoreOf<PreAuthorizationReview>) -> some SwiftUI.View {
+			ScrollView(showsIndicators: false) {
+				VStack(spacing: .zero) {
+					header(dAppMetadata: viewStore.dAppMetadata)
+
+					Group {
+						if let manifest = viewStore.displayMode.rawManifest {
+							rawManifest(manifest)
+						} else {
+							details(viewStore.showRawManifestButton)
+						}
+					}
+					.background(Common.gradientBackground)
+					.clipShape(RoundedRectangle(cornerRadius: .small1))
+					.padding(.horizontal, .small2)
+
+					feesInformation(dAppName: viewStore.dAppName)
+						.padding(.top, .small2)
+						.padding(.horizontal, .small2)
+
+					expiration(viewStore.expiration, secondsToExpiration: viewStore.secondsToExpiration)
+
+					ApprovalSlider(
+						title: L10n.PreAuthorizationReview.slideToSign,
+						resetDate: viewStore.sliderResetDate
+					) {}
+						.controlState(viewStore.sliderControlState)
+						.padding(.horizontal, .medium2)
+				}
+				.animation(.easeInOut, value: viewStore.displayMode.rawManifest)
+			}
+			.coordinateSpace(name: coordSpace)
+			.onPreferenceChange(PositionsPreferenceKey.self) { positions in
+				guard let offset = positions[navTitleID]?.maxY else {
+					showNavigationTitle = true
+					return
+				}
+				if showNavigationTitle, offset > showTitleHysteresis {
+					showNavigationTitle = false
+				} else if !showNavigationTitle, offset < 0 {
+					showNavigationTitle = true
+				}
+			}
```

Comment:
Maybe this could be factored out as a view builder along with:
```swift
@SwiftUI.State private var showNavigationTitle = false

		private let coordSpace: String = "PreAuthorizationReviewCoordSpace"
		private let navTitleID: String = "PreAuthorizationReview.title"
		private let showTitleHysteresis: CGFloat = .small3

```

defined at top. Could also add the `header` there.

Reply:
Attempted this [here](https://github.com/radixdlt/babylon-wallet-ios/pull/1371/commits/0be24a389c9582c057ae25b8a4b4ae1c3c3ea63b), but ended up reverting it due to some issues with the shadow of the top JaggedEdge, as seen below.

Even if it is solved, I am not sure if it is a great approach since the child view (`VisibleHeaderView`), would be setting the `toolbar`, and I've seen some weird behavior in cases like this.

| Before | After |
| -- | -- |
| <img src=https://github.com/user-attachments/assets/a6219b66-6c5c-4265-a88c-8ace4484ef9f width=200> | <img src=https://github.com/user-attachments/assets/bd0a81ca-00e9-4a1f-b3fe-494b86649e7d width=200> |

Reply:
sure, fair enough, we don't gain a lot with duplication removal